app/views/posts/index.html.erb

```diff
@@ -1,31 +1,12 @@
-<div class="posts-container">
-<div class="users">
-<%= render 'shared/user', user:@user %>
-</div>
-
-<div class="posts">
-<h2>All Posts</h2>
-
-<%=link_to "Add post", new_user_post_path(@user)%>
-
-<% @posts.each do |post| %>
-<%= render 'shared/post', post: post, user:@user %>
-
-<div class="post-content">
-<%= link_to "Add Comment", new_user_post_comment_path(@user, post) %>
-
-<%= form_with url: user_post_likes_path(post.author.id, post.id) do |form| %>
-
-<%= form.submit 'Like' %>
+<% if user_signed_in? %>
+<div> Welcome <%= current_user.email %> </div>
+<%= button_to "Sign out", destroy_user_session_path, method: :delete %>
+<% else %>
+<%= button_to "Sign in", new_user_session_path %>
 <% end %>
-</div>
 
-<% if post.recent_comments.present? %>
-<% comments = post.recent_comments %>
-<%= render "shared/comments", comments:comments %>
-<% end %>
+<div class="users">
+<% @users.each do |user| %>
+<%= render 'shared/user', user:user %>
```

Comment:
- [ ] Kindly fix this error as @users returns nil.
![image](https://github.com/firstpj/BLOG-appllication/assets/91486791/eb351f65-4655-4d65-9524-033914cc842d)